src/core/transaction.c

```diff
@@ -400,7 +400,7 @@ static int transaction_verify_order_one(Transaction *tr, Job *j, Job *from, unsi
                                                     "Found %s on %s/%s",
                                                     unit_id == array ? "ordering cycle" : "dependency",
                                                     *unit_id, *job_type),
-                                   "%s", unit_ids);
+                                   "%s", strna(unit_ids));
```

Comment:
Wouldn't `strnull()` be better? I think we should use that instead to generally signify that we're not printing something because of an allocation failure.

Reply:
As suggested at https://github.com/systemd/systemd/pull/26331#discussion_r1128429650, we should introduce stroom() or so, to indicate OOM error. I'd like to keep that as is in this PR. Let's handle OOM error in logging more consistently in another PR.